com.playeveryware.eos/Runtime/Core/Config/Attributes/ConfigFieldAttribute.cs

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using System;
+
+    [AttributeUsage(AttributeTargets.Field)]
+    public class ConfigFieldAttribute : Attribute
```

Comment:
Would there be benefit in making this abstract?

Reply:
No, because the ConfigFieldAttribute class itself is used in many circumstances (actually in most of them):

Here are some assorted examples from the `development` branch.

https://github.com/PlayEveryWare/eos_plugin_for_unity/blob/2a18ec7e3fddff1c35baf37195a9d7989be1c4bd/Assets/Plugins/Source/Editor/Configs/ToolsConfig.cs#L59

https://github.com/PlayEveryWare/eos_plugin_for_unity/blob/2a18ec7e3fddff1c35baf37195a9d7989be1c4bd/Assets/Plugins/Source/Editor/Configs/LibraryBuildConfig.cs#L36

https://github.com/PlayEveryWare/eos_plugin_for_unity/blob/2a18ec7e3fddff1c35baf37195a9d7989be1c4bd/Assets/Plugins/Source/Editor/Configs/SigningConfig.cs#L51

It's worth noting that C# will _specifically for attribute classes_ allow you to omit the `Attribute` component of the class name when decorating things with your custom attribute. This can be a point of confusion for people who are less familiar with attributes.